### PR TITLE
Install unit testing

### PR DESCRIPTION
**Pull request by charliew-rsi**
This branch is based off register-validation and includes install-immutability-helper. This branch resolves dependency conflicts and provides a blank template for doing unit testing with Jest and Enzyme.

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
```
Comment (wesdollar) [inline src/setupTests.js]: New line at end of the file

File: src/__tests__/components/example.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import Register from '../../components/Register';
+import { shallow } from 'enzyme';
+import configureStore from "redux-mock-store";
+
+describe("Renders Register component", () => {
+    const mockStore = configureStore();    
+    let wrapper;
+    
+    it('renders the form', () => {
+        const wrapper = shallow(<Register />);
```
[the remark this answers is not in the document]
Comment (charliew-rsi) [reply src/__tests__/components/example.test.js]: resolved

File: src/components/Register.js [hunk 3]
```diff
@@ -1,53 +1,100 @@
...
       ev.preventDefault();
-      this.props.onSubmit(username, email, password);
+      agent.Auth.register(username, email, password);
     }
+
+    this.changeFullName = this.changeFullName.bind(this);
+    this.changeEmail = this.changeEmail.bind(this);
+    this.changeUsername = this.changeUsername.bind(this);
+    this.changePassword = this.changePassword.bind(this);
+    this.checkForErrors = this.checkForErrors.bind(this);
```
[the remark this answers is not in the document]
Comment (charliew-rsi) [reply src/components/Register.js]: resolved

File: src/components/Register.js [hunk 4]
```diff
@@ -1,138 +1,198 @@
...
+        lastName: /^\D+$/,
+        username: /^\w{5,}$/,
+        password: /^.{8,}$/,
+        email: /^[\w.]+@\w+.\w+$/
+      }
+    };
+
+    this.isDisabled = this.isDisabled.bind(this);
 
     this.submitForm = (username, email, password) => ev => {
```
Comment (wesdollar) [inline src/components/Register.js]: Please see comments for this file on other branch that has already been reviewed. Make sure this branch includes those changes.

**Pull request closed without merging**